src/envoy/mixer/http_control.cc

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/envoy/mixer/http_control.h"
+#include "common/common/utility.h"
+#include "common/http/utility.h"
+
+using ::google::protobuf::util::Status;
+using ::istio::mixer_client::Attributes;
+using ::istio::mixer_client::DoneFunc;
+
+namespace Http {
+namespace Mixer {
+namespace {
+
+const std::string kProxyPeerID = "Istio/Proxy";
+const std::string kEnvNameTargetService = "TARGET_SERVICE";
+
+// Define lower case string for X-Forwarded-Host.
+const LowerCaseString kHeaderNameXFH("x-forwarded-host", false);
+
+// Define attribute names
+const std::string kAttrNamePeerId = "peerId";
+const std::string kAttrNameURL = "url";
+const std::string kAttrNameHttpMethod = "httpMethod";
```

Comment:
I added all http headers.  It looks like these

HTTP-request-header::authority: (STRING): localhost:9090
HTTP-request-header::method: (STRING): POST
HTTP-request-header::path: (STRING): /echo?key=API-KEY
HTTP-request-header:accept: (STRING): */*
HTTP-request-header:content-length: (STRING): 11
HTTP-request-header:content-type: (STRING): application/x-www-form-urlencoded
HTTP-request-header:user-agent: (STRING): curl/7.35.0
HTTP-request-header:x-forwarded-proto: (STRING): http
HTTP-request-header:x-request-id: (STRING): fe6cd0cf-31b6-40c0-b352-3b5186daec2c
HTTP-request-param:key: (STRING): API-KEY
HTTP-response-header::status: (STRING): 200
HTTP-response-header:content-length: (STRING): 11
HTTP-response-header:content-type: (STRING): application/x-www-form-urlencoded
HTTP-response-header:date: (STRING): Sat, 04 Feb 2017 00:22:44 GMT
HTTP-response-header:server: (STRING): envoy
HTTP-response-header:x-envoy-upstream-service-time: (STRING): 0
peerId: (STRING): Istio/Proxy
requestSize: (INT64): 11
responseSize: (INT64): 11
responseTime: (INT64): 3
targetService: (STRING): abdcecee